### PR TITLE
feat(menu/player-modal): Allow usernames in txadmin command

### PR DESCRIPTION
**Pull request by Poggicek**
/tx or /txadmin will now accept usernames with ids, if multiple clients with similiar names are found it will fill out the search input with the arguments from txadmin command and list all similiar names. Unless the search input is modified it will be cleared when the menu gets closed.

### CODE REVIEW

File: locale/en.json [hunk 1]
```diff
@@ -179,7 +179,7 @@
         "player_modal": {
             "misc": {
                 "error": "An error occurred fetching this users details. The error is shown below:",
-                "target_not_found": "Was unable to find an online player with ID %{targetId}"
+                "target_not_found": "Was unable to find an online player with ID %{target}"
```
Comment (TasoOneAsia) [inline locale/en.json]: Since this can now be a name or ID this phrase won't work for a name.

File: menu/src/MenuWrapper.tsx [hunk 2]
```diff
@@ -39,9 +43,16 @@ const MenuWrapper: React.FC = () => {
...
       setPage(txAdminMenuPage.Main);
     }, 750);
+
+    if (playersFilterIsTemp) {
+      setPlayerFilter("");
+      setPlayersFilterIsTemp(false);
+    }
+
     return () => clearInterval(changeTimer);
   }, [visible]);
```
Comment (TasoOneAsia) [inline menu/src/MenuWrapper.tsx]: Missing dependent state in dep array. The setter function is stable and can be ignored from deps.

```suggestion
  }, [visible, playersFilterIsTemp]);
```

File: menu/src/hooks/useHudListenersService.tsx [hunk 3]
```diff
@@ -115,14 +117,32 @@ export const useHudListenersService = () => {
...
+    if (targetId) {
+      targetPlayer = onlinePlayers.find(
+        (playerData) => playerData.id === targetId
+      );
+    } else {
+      const foundPlayers = onlinePlayers.filter(
+        (playerData) => playerData.username.toLowerCase().includes(target.toLowerCase())
+      );
+
+      if (foundPlayers.length == 1)
```
Comment (TasoOneAsia) [inline menu/src/hooks/useHudListenersService.tsx]:
```suggestion
      if (foundPlayers.length === 1)
```

File: scripts/menu/client/cl_base.lua [hunk 4]
```diff
@@ -22,10 +22,10 @@ local function txadmin(_, args)
   end
 
   if menuIsAccessible then
-    local targetPlayer = args[1] and tonumber(args[1])
     toggleMenuVisibility()
     -- Shortcut to open a specific players profile
-    if isMenuVisible and targetPlayer then
+    if isMenuVisible and #args >= 1 then
```
Comment (TasoOneAsia) [inline scripts/menu/client/cl_base.lua:27]: Guess this is being done to handle names with spaces in them?

File: menu/src/hooks/useHudListenersService.tsx [hunk 5]
```diff
@@ -115,14 +117,32 @@ export const useHudListenersService = () => {
   });
 
   // Handler for dynamically opening the player page & player modal with target
-  useNuiEvent<number>("openPlayerModal", (targetId) => {
-    const targetPlayer = onlinePlayers.find(
-      (playerData) => playerData.id === targetId
-    );
+  useNuiEvent<string>("openPlayerModal", (target) => {
+    let targetPlayer
+    let targetId = parseInt(target)
```
Comment (TasoOneAsia) [inline menu/src/hooks/useHudListenersService.tsx]: Not reassigned anywhere.
```suggestion
    const targetId = parseInt(target)
```

File: menu/src/state/players.state.ts [hunk 6]
```diff
@@ -83,6 +83,12 @@ const playersState = {
     key: "filterPlayerDataInput",
     default: "",
   }),
+
```
Comment (TasoOneAsia) [inline menu/src/state/players.state.ts]:
```suggestion
```

**Review by TasoOneAsia — CHANGES_REQUESTED** [review]
Before in-game testing

File: scripts/menu/client/cl_base.lua (hunk 4 above)
Comment (Poggicek) [reply scripts/menu/client/cl_base.lua:27]: yes

File: locale/en.json [hunk 7]
```diff
@@ -179,7 +179,7 @@
         "player_modal": {
             "misc": {
                 "error": "An error occurred fetching this users details. The error is shown below:",
-                "target_not_found": "Was unable to find an online player with ID %{targetId}"
+                "target_not_found": "Was unable to find an online player with ID or an username %{target}"
```
Comment (TasoOneAsia) [inline locale/en.json]:
```suggestion
                "target_not_found": "Was unable to find an online player with ID or an username of %{target}"
```

**Review by TasoOneAsia — APPROVED** [review]
Small change and all good from me.

**Comment by tabarra** [comment]
You guys didn't update the chat suggestion? 🤦‍♂️

**Comment by TasoOneAsia** [comment]
Apparently not 🤔, oh well

**Pull request merged**